Dependency Update: mocha/rewire

tzfx opened the pull request:
- Mocha 7 -> 10
- Removal of mochawesome, which we didn't appear to be using for anything.
- Upgrade of rewire to 6.

**Down to 11 vulnerabilities.**

This update jumps us from Mocha 7 to 10. All tests and reporting appears to have completed without issue despite not actually having to update any tests.

Handlebars is throwing some errors in the email templating for teams, though I verified that the body is properly templated-- so I'm not sure what's going on with that.

Handlebars and mongoose still need to be updated any way to mitigate vulnerabilities.

Commit:
deps: upgraded rewrite.

Commit:
deps: mocha 10

Commit:
deps: yargs?

jrassa commented:
Looks like we could probably just remove rewire.  It is only used to test `paging-search.plugin.js` which we deprecated a while back in favor more discrete plugins.

tzfx commented:
Added yargs dep and reworked typing a bit to get tsc to pass. There's an open issue with yargs' argv typing (`{} | Promise<{}>`) that makes tsc break without an await-- of which you can't provide without node15+... :P Added a comment and an `any` coercion to get tsc to pass.

All tests and all flags still work.

tzfx commented on src/test.js:
```diff
@@ -1,11 +1,25 @@
+ * @type {any}
+ * This coercion is required due to yarg's unfortunate split {} | Promise<{}> typing.
+ * TSC tries to reconcile the two types and is unable to do so.
+ * @see https://github.com/yargs/yargs/issues/2175ß
```
👀
